src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -2063,6 +2076,28 @@ private void ShowOpenDialogAndOpenResult(object parameter)
                     else
                     {
                         Open(new Tuple<string, bool>(_fileDialog.FileName, _fileDialog.RunManualMode));
+                        LastSavedLocation = Path.GetDirectoryName(_fileDialog.FileName);
+                    }
+                }
+            }
+        }
+
+        private void SetDefaultInitialDirectory(DynamoOpenFileDialog _fileDialog)
+        {
+            //check if the last accessed path was the templates directory, if yes, change it to default
```

Comment:
Can we use try-catch exception for the code here, as file path APi's are used here.